src/strategy_manager/strategy_manager.cpp

```diff
@@ -0,0 +1,156 @@
+/******************************************************************************
+ * Copyright © 2013-2019 The Komodo Platform Developers.                      *
+ *                                                                            *
+ * See the AUTHORS, DEVELOPER-AGREEMENT and LICENSE files at                  *
+ * the top-level directory of this distribution for the individual copyright  *
+ * holder information and the developer policies on copyright and licensing.  *
+ *                                                                            *
+ * Unless otherwise agreed in a custom licensing agreement, no part of the    *
+ * Komodo Platform software, including this file may be copied, modified,     *
+ * propagated or distributed except according to the terms contained in the   *
+ * LICENSE file                                                               *
+ *                                                                            *
+ * Removal or modification of this copyright notice is prohibited.            *
+ *                                                                            *
+ ******************************************************************************/
+
+#include <vector>
+#include <unordered_map>
+
+#include "strategy_manager.hpp"
+
+namespace antara {
+  bool asset::operator==(const asset &other) const {
+    return symbol == other.symbol;
+  };
+
+  std::pair<antara::asset, antara::asset> pair::to_std_pair() {
+    return std::pair<antara::asset, antara::asset>(base, quote);
+  };
+
+  bool pair::operator==(const pair &rhs) const {
+    return base == rhs.base && quote == rhs.quote;
+  };
+}
+
+namespace std {
+
+  template <>
+  struct hash<antara::pair> {
+    std::size_t operator()(const antara::pair& p) const {
+      using std::size_t;
+      using std::hash;
+      using std::string;
+
+      std::size_t h1 = std::hash<std::string>{}(p.base.symbol);
+      std::size_t h2 = std::hash<std::string>{}(p.quote.symbol);
+
+      return h1 ^ (h2 << 1);
+    }
+  };
+}
+
+enum class Side { BUY, SELL, BOTH };
```

Comment:
side

Reply:
enum contents in lower case please

Reply:
Making this changes, I get an error:

```
src/strategy_manager/strategy.manager.cpp:60:8: error: declaration of ‘side market_making_strategy::side’ [-fpermissive]
   side side;
```

Can I not have a variable and a type with the same name?

Reply:
you can use namespace mynamespace::my_type variable_name;

for private member of a class the fields should be: my_type field_name_

Reply:
so you can store 'antara::side side_';